scripts/verify_commits

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+function err() {
+    echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $*" >&2
+}
+
+function verify_staging_diff() {
+    local remote="${1}"
+    local upstream_commit="${2}"
+    local downstream_commit="${3}"
+    local staging_dir="staging/${remote}"
+
+    local outside_staging
+    outside_staging="$(git show --name-only "${downstream_commit}" -- ":!${staging_dir}")"
+    if [[ -n "${outside_staging}" ]]; then
+        err "downstream staging commit ${downstream_commit} changes files outside of ${staging_dir}:"
+        err "${outside_staging}"
+        err "factor out changes to ${staging_dir} into a standalone commit"
+        return 1
+    fi
+
+    local divergence
+    divergence="$(git diff --stat "${upstream_commit}" "${downstream_commit}:${staging_dir}" -- ':!vendor')"
+    if [[ -n "${divergence}" ]]; then
+        err "downstream staging commit ${downstream_commit} diverges from upstream commit ${upstream_commit}:"
+        err "${divergence}"
+        err "changes must match referenced upstream commit ${upstream_commit}"
+        return 1
+    fi
```

Comment:
should I remove this? it might cause problems if we need to make some changes to accommodate diffs in staging due to any downstream-only changes to staging.

Reply:
won't this explicitly break what we're doing now with manifest changes? we are amending commits to do that right now so the changes are as idempotent as possible

at the very least, we need to ignore vendor changes as well